encfs/FileUtils.cpp

```diff
@@ -1073,28 +1081,32 @@ RootPtr createV6Config(EncFS_Context *ctx,
     keySize = selectKeySize(alg);
     blockSize = selectBlockSize(alg);
     nameIOIface = selectNameCoding();
-    if (reverseEncryption) {
-      cout << _("reverse encryption - chained IV and MAC disabled") << "\n";
-      uniqueIV = selectUniqueIV(false);
-      /* Reverse mounts are read-only by default (set in main.cpp).
-       * If uniqueIV is off, writing can be allowed, because there
-       * is no header that could be overwritten */
-      if (uniqueIV == false) opts->readOnly = false;
-    } else {
+
+    chainedIV = selectChainedIV();
+    uniqueIV = selectUniqueIV(true);
+    if (chainedIV && uniqueIV)
+      externalIV = selectExternalChainedIV();
+    else {
+          // xgroup(setup)
+          cout << _("External chained IV disabled, as both 'IV chaining'\n"
+                    "and 'unique IV' features are required for this option.")
+               << "\n";
+          externalIV = false;
+    }
+    selectBlockMAC(&blockMACBytes, &blockMACRandBytes, opts->requireMac, reverseEncryption);
+
+    /* Reverse mounts are read-only by default (set in main.cpp).
+     * If uniqueIV is off, writing can be allowed, because there
+     * is no header that could be overwritten */
+    if (uniqueIV == false && blockMACBytes == 0) opts->readOnly = false;
       chainedIV = selectChainedIV();
```

Comment:
Indenting here is a little misleading because the prior conditional has the statement on the same line.